packages/web/src/pages/sign-in-page/SignInWithMetaMaskButton.tsx

```diff
@@ -1,42 +1,122 @@
-import { useCallback } from 'react'
+import { useCallback, useState } from 'react'
 
-import { route } from '@audius/common/utils'
-import { Button, ButtonProps, IconMetamask } from '@audius/harmony'
+import { Status } from '@audius/common/models'
+import { FEED_PAGE, SIGN_UP_EMAIL_PAGE } from '@audius/common/src/utils/route'
+import { accountActions } from '@audius/common/store'
+import {
+  Button,
+  ButtonProps,
+  IconMetamask,
+  Text,
+  TextLink
+} from '@audius/harmony'
+import { metaMask } from '@wagmi/connectors'
+import { disconnect } from '@wagmi/core'
+import { useDispatch } from 'react-redux'
+import { useAccount, useConnect, useSwitchChain } from 'wagmi'
 
+import { getRouteOnCompletion } from 'common/store/pages/signon/selectors'
 import { useNavigateToPage } from 'hooks/useNavigateToPage'
+import { doesUserExist } from 'pages/sign-up-page/components/ExternalWalletSignUpModal'
 import { userHasMetaMask } from 'pages/sign-up-page/utils/metamask'
-
-const { FEED_PAGE } = route
+import { initSdk } from 'services/audius-sdk'
+import { audiusChain, wagmiConfig } from 'services/audius-sdk/wagmi'
+import { reportToSentry } from 'store/errors/reportToSentry'
+import { useSelector } from 'utils/reducer'
 
 const messages = {
-  signIn: 'Sign In With MetaMask'
+  signIn: 'Sign In With MetaMask',
+  noAccountError: () => (
```

Comment:
new UI experience. Before, this would just let you attempt the sign in which would simply refresh the page and you'd still be logged out